File: plugins/de.ovgu.featureide.examples/featureide_examples/GPL-FH-JML/features/Connected/GPL/Graph.java
package GPL;

// *****************************************************************
   
public class Graph 
{
    // Executes Connected Components
	/*@requires \original;
	 ensures \original;
	 assignable \nothing;
	 @*/
    public void run( Vertex s )
    {
	     	System.out.println("Connected");
        ConnectedComponents( );
        original( s );
    }

    public void ConnectedComponents( ) 
    {
        GraphSearch( new RegionWorkSpace( ) );
    }
}

File: plugins/de.ovgu.featureide.examples/featureide_examples/GPL-FH-CSharp/features/Number/Graph.cs

// ***********************************************************************
   
public class Graph //refines
{
    // Executes Number Vertices
    public void run( Vertex s )
     {
       	System.Console.Out.WriteLine("Number");
        NumberVertices( );
        original( s );
    }

    public void NumberVertices( ) 
    {
        GraphSearch( new NumberWorkSpace( ) );
    }
}
